src/main/kotlin/io/github/soat7/myburguercontrol/domain/service/PaymentService.kt

```diff
@@ -1,30 +1,35 @@
 package io.github.soat7.myburguercontrol.domain.service
 
 import io.github.soat7.myburguercontrol.application.ports.inbound.PaymentServicePort
+import io.github.soat7.myburguercontrol.application.ports.outbound.PaymentDatabasePort
 import io.github.soat7.myburguercontrol.application.ports.outbound.PaymentIntegrationPort
 import io.github.soat7.myburguercontrol.domain.enum.PaymentStatus
+import io.github.soat7.myburguercontrol.domain.model.Order
 import io.github.soat7.myburguercontrol.domain.model.Payment
 import mu.KLogging
 
 class PaymentService(
-    private val paymentIntegrationPort: PaymentIntegrationPort
+    private val paymentIntegrationPort: PaymentIntegrationPort,
+    private val paymentDatabasePort: PaymentDatabasePort
 ) : PaymentServicePort {
 
     private companion object : KLogging()
 
-    override fun requestPayment(payment: Payment): Payment {
-        logger.info { "Starting to request payment integration with id: [${payment.id}]" }
+    override fun requestPayment(order: Order): Payment {
+        logger.info { "Starting to request payment integration for order id: [${order.id}]" }
 
-        val paymentResult = paymentIntegrationPort.requestPayment(payment)
+        val paymentResult = paymentIntegrationPort.requestPayment(order)
 
-        payment.apply {
-            this.authorizationId = paymentResult.authorizationId
-            this.status = checkApproval(paymentResult.approved)
+        order.payment.apply {
+            authorizationId = paymentResult.authorizationId
+            status = checkApproval(paymentResult.approved)
         }
 
-        logger.info { "Successfully integrated with status return: [${payment.status.name}]" }
+        paymentDatabasePort.create(order.payment)
 
-        return payment
+        logger.info { "Successfully integrated with status return: [${order.payment.status.name}]" }
```

Comment:
Colocaria logs mais detalhados, checando o status do retorno. Tendo um logger.warn ou logger.error quando o pagamento não for aprovado.

Reply:
vou melhorar os logs apos o MR